### ISSUE TITLE
[BUG] S3 Sink Avro Output issues

### ISSUE THREAD

**Issue by engechas** [opensearch-project/data-prepper#3158]
**Describe the bug**
A few bugs found for S3 Sink Avro output codecs:
1. With `compression: gzip` specified, the files are still uploaded with just a `.avro` suffix. This should be `.avro.gz`
2. The contents of the avro file show null for all fields:
```
{"version":null,"srcport":null,"dstport":null,"accountId":null,"interfaceId":null,"srcaddr":null,"dstaddr":null,"start":null,"end":null,"protocol":null,"packets":null,"bytes":null,"action":null,"logStatus":null}
{"version":null,"srcport":null,"dstport":null,"accountId":null,"interfaceId":null,"srcaddr":null,"dstaddr":null,"start":null,"end":null,"protocol":null,"packets":null,"bytes":null,"action":null,"logStatus":null}
{"version":null,"srcport":null,"dstport":null,"accountId":null,"interfaceId":null,"srcaddr":null,"dstaddr":null,"start":null,"end":null,"protocol":null,"packets":null,"bytes":null,"action":null,"logStatus":null}
{"version":null,"srcport":null,"dstport":null,"accountId":null,"interfaceId":null,"srcaddr":null,"dstaddr":null,"start":null,"end":null,"protocol":null,"packets":null,"bytes":null,"action":null,"logStatus":null}
```
3. The timestamp in the file name is 12 hours behind UTC. I am not sure what the "correct" time is but UTC seems logical

**To Reproduce**
Sink config:
```
sink:
    - s3:
        aws:
          region: "us-west-2"
          sts_role_arn: "<my role>"
        bucket: "my-sink-bucket"
        object_key:
          path_prefix: "s3-sink"
        threshold:
          event_collect_timeout: 600s
        compression: "gzip"
        codec:
          avro:
            schema: >
              {
                "type" : "record",
                "namespace" : "org.opensearch.dataprepper.examples",
                "name" : "VpcFlowLog",
                "fields" : [
                  { "name" : "version", "type" : ["null", "string"]},
                  { "name" : "srcport", "type": ["null", "int"]},
                  { "name" : "dstport", "type": ["null", "int"]},
                  { "name" : "accountId", "type" : ["null", "string"]},
                  { "name" : "interfaceId", "type" : ["null", "string"]},
                  { "name" : "srcaddr", "type" : ["null", "string"]},
                  { "name" : "dstaddr", "type" : ["null", "string"]},
                  { "name" : "start", "type": ["null", "int"]},
                  { "name" : "end", "type": ["null", "int"]},
                  { "name" : "protocol", "type": ["null", "int"]},
                  { "name" : "packets", "type": ["null", "int"]},
                  { "name" : "bytes", "type": ["null", "int"]},
                  { "name" : "action", "type": ["null", "string"]},
                  { "name" : "logStatus", "type" : ["null", "string"]}
                ]
              }
```

**Screenshots**
File name times example:
![image](https://github.com/opensearch-project/data-prepper/assets/62891993/045dbc4b-847f-4c66-9b36-d4cc0f834d38)

**Comment by dlvenable** [comment]
The Avro codec does not work with union types, nor is there any support for it in the code. https://github.com/opensearch-project/data-prepper/blob/60f69b3c565e7a10e6459d6bbe694d74ae6e6d96/data-prepper-plugins/avro-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/avro/AvroOutputCodec.java#L192-L227

**Comment by engechas** [comment]
For the timestamp in the file name, the same behavior is seen with ndjson so it doesn't look related to avro, rather S3 sink as a whole

**Comment by engechas** [comment]
#3171 fixes number 3

**Comment by dlvenable** [comment]
PR #3194 fixes item 2.

**Comment by dlvenable** [comment]
Item 1 should be resolved by #3196.
